src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -322,27 +322,34 @@ class MigrationTo0_13(taskRepository: TaskRepository, store: PersistentStore) {
   def migrateTasks(): Future[Unit] = {
     log.info("Start 0.13 migration")
 
-    // including 0.12, task keys are in format task:app:app.uuid – the taskId is
-    // already contained the appId. When using the generic EntityRepo, a colon
-    // in the key after the prefix implicitly denotes a versioned entry, so this
-    // had to be changed, even though tasks are not stored with versions.
-    def migrateKey(legacyKey: String): Future[Unit] = {
-      fetchLegacyTask(legacyKey).flatMap {
-        case Some(task) => taskRepository.store(task).flatMap { _ =>
-          entityStore.expunge(legacyKey).map(_ => ())
-        }
-        case _ => Future.failed[Unit](new RuntimeException(s"Unable to load entity with key = $legacyKey"))
-      }
-    }
-
     entityStore.names().flatMap { keys =>
-      log.info("Found {} tasks to migrate", keys.size)
-      Future.sequence(keys.map(migrateKey)).map(_ => ())
+      log.info("Found {} tasks in store", keys.size)
+      // old format is appId:appId.taskId
+      val oldFormatRegex = """^.*:.*\..*$""".r
+      val namesInOldFormat = keys.filter(key => oldFormatRegex.pattern.matcher(key).matches)
+      log.info("{} tasks in old format need to be migrated.", namesInOldFormat.size)
+
+      namesInOldFormat.foldLeft(Future.successful(())) { (f, nextKey) =>
+        f.flatMap(_ => migrateKey(nextKey))
+      }
     }.map { _ =>
       log.info("Completed 0.13 migration")
     }
   }
 
+  // including 0.12, task keys are in format task:app:app.uuid – the taskId is
```

Comment:
Nit-picking: An example would be nice. I think the "task:" prefix is literal and "app" and "app.uuid" are place-holders for some actual values. An example would clarify that.